src/meta/src/rpc/ddl_controller.rs

```diff
@@ -750,8 +750,6 @@ impl DdlController {
             .try_into()
             .expect("Target table should exist in sink into table");
 
-        assert_eq!(table_catalog.incoming_sinks, target_table.incoming_sinks);
```

Comment:
The `incoming_sinks` in table catalog could be fixed during recovery when issuing a `drop` command, we should return error rather than panic here.